### PR TITLE
CRYPTO-93: add a factory method to provide simpler API

### PR DESCRIPTION
**Pull request by kexianda**
[CRYPTO-93](https://issues.apache.org/jira/browse/CRYPTO-93)
 use the better hardware random by default

### CODE REVIEW

File: src/main/java/org/apache/commons/crypto/examples/RandomExample.java [hunk 1]
```diff
@@ -32,29 +32,62 @@
...
+
+            //Closes the CryptoRandom.
+            random.close();
+
+            // Show the generated output
+            System.out.println(Arrays.toString(key));
+
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/crypto/examples/RandomExample.java]: This is sample code so should be as simple as possible.
Drop the try/catch and throw Exception from the method instead

File: src/main/java/org/apache/commons/crypto/examples/RandomExample.java [hunk 2]
```diff
@@ -32,29 +32,62 @@
...
+            System.out.println(Arrays.toString(key));
+            System.out.println(Arrays.toString(iv));
 
-        // Show the generated output
-        System.out.println(Arrays.toString(key));
-        System.out.println(Arrays.toString(iv));
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
     }
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/crypto/examples/RandomExample.java]: Ditto; don't try/catch - throw instead

File: src/main/java/org/apache/commons/crypto/random/CryptoRandomFactory.java [hunk 3]
```diff
@@ -134,6 +134,21 @@ public static CryptoRandom getCryptoRandom(Properties props)
     }
 
     /**
+     * Get a OpenSSL CryptoRandom by default.
+     * If OpenSSL is unavailable qand fallback is enable, we will use Java Random
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/crypto/random/CryptoRandomFactory.java]: Fallback no longer exists

File: src/main/java/org/apache/commons/crypto/examples/RandomExample.java [hunk 4]
```diff
@@ -32,29 +32,62 @@
...
-        byte[] iv = new byte[32];
+        exampleWithdefaultHardwareRandom();
+        exampleWithUserDefinedRandom();
+    }
+
+    public static void exampleWithdefaultHardwareRandom() {
+        try {
+            //Constructs a byte array to store random data.
+            byte[] key = new byte[16];
+
```
Comment (sebbASF) [inline src/main/java/org/apache/commons/crypto/examples/RandomExample.java]: This does not agree with the other example as it only uses a single byte array

**Comment by kexianda** [comment]
thanks @sebbASF for your comments. make sense. updated.

**Pull request closed without merging**